parserasgeo/prg.py

```diff
@@ -150,6 +153,46 @@ def return_xs(self, xs_id, river, reach, strip=False, rnd=False, digits=0):
                 if test_xs_id == wanted_xs_id and test_river == wanted_river and test_reach == wanted_reach:
                     return item
         raise CrossSectionNotFound
+        
+    def return_culvert(self, culvert_id, river, reach, strip=False, rnd=False, digits=0):
```

Comment:
return_culvert() and return_xs() are identical except for the type of feature (class) used to compare against in new line 181 (      if isinstance(item, Culvert): ).  Create method _return_node() which also accepts a feature type (CrossSection, Culvert, etc) and is called by wrappers return_culvert() and return_xs().